.github/workflows/lint.yml

```diff
@@ -11,11 +11,27 @@ jobs:
     # We need to run the linter on the same image we use for building, since it
     # needs the C libraries installed for the dependencies to typecheck.
     runs-on: ubuntu-latest
-    container: cometbft/cometbft-db-testing
     steps:
       - uses: actions/checkout@v4
-      - uses: golangci/golangci-lint-action@v4.0.0
-        with:
-          args: --timeout 10m
-          version: latest
-          github-token: ${{ secrets.github_token }}
+      runs-on: ubuntu-latest
```

Comment:
What's the reason for running it in Docker?

Reply:
make it similar to the tests.  It was run in docker previously, too.  We could also run it outside of docker, but we would need to change the pattern where we use build tags to specify which DB we are using.  

If you'd like to do that, and reduce DBs to only those that work I can whip up that PR with ease.

Reply:
...sir, I am going to just do that PR.  It is really the sensible way to cut down this repository to the only two reasonable options, and we've done the user surveys